src/worker_pool_sup.erl

```diff
@@ -28,26 +28,29 @@
 
 -spec(start_link/0 :: () -> rabbit_types:ok_pid_or_error()).
 -spec(start_link/1 :: (non_neg_integer()) -> rabbit_types:ok_pid_or_error()).
+-spec(start_link/2 :: (non_neg_integer(), atom())
+                   -> rabbit_types:ok_pid_or_error()).
 
 -endif.
 
 %%----------------------------------------------------------------------------
 
--define(SERVER, ?MODULE).
-
-%%----------------------------------------------------------------------------
-
 start_link() ->
     start_link(erlang:system_info(schedulers)).
 
 start_link(WCount) ->
-    supervisor:start_link({local, ?SERVER}, ?MODULE, [WCount]).
+    start_link(WCount, worker_pool).
```

Comment:
we should probably have a header which defines the default worker pool name rather than repeating it here. see also this line in `worker_pool.erl`:
```erlang
-define(DEFAULT_SERVER, ?MODULE).
```